src/contents/en-gb/doc_android.md

```diff
@@ -0,0 +1,255 @@
+
+```
+if (entry.getValue().getContentLen() != 0)
+```
+
+To delete an entry in the mutable data, we can use the delete entry action by passing the key of the entry to be deleted. Note, that the delete action also requires the `entryVersion + 1` as a parameter. The delete action invokes the `deleteEntry()` function in the `SafeApi` class.
+
+```
+final NativeHandle actionHandle = session.mDataEntryAction.newEntryAction().get();
+session.mDataEntryAction.delete(actionHandle, key, version).get();
+session.mData.mutateEntries(mDataInfo, actionHandle).get();
+```
+
+Here, the version increment is done in the [`SafeTodoService.deleteTask()`](https://github.com/maidsafe/safe-getting-started-android/blob/master/app/src/main/java/net/maidsafe/sample/services/SafeTodoService.java#L124) function *before* calling the Safe API.
+
+## Making use of the App's Container
+
+Since the `appData` we work with is randomly generated, a different `appData` is generated every time the app starts and consequently the entries made in the previous mutable data will be lost. Thus, we need to store this `appData` in a location that is accessible by the application. For this, we can use the App's Container. The app's container is mutable data that is exclusive per app per user. To obtain the `appData` info for the application:
+```
+MDataInfo appContainerInfo = session.getContainerMDataInfo(APP_CONTAINER_NAME + appId).get();
+MDataValue mDataValue = session.mData.getValue(appContainerInfo, key).get();
+info  = session.mData.deserialise(mDataValue.getContent()).get();
+```
+
+We can serialize and store the `appData` in the App Container so that we can re-use the mutable data that was randomly generated the first time the app is launched.
+
+So, when the application starts we should first check for existent `appData` in the App's Container, fetch it and deserialize it. If the data does not exist, then we create a new mutable data and store it in the app's container. This logic is implemented in the [`getSectionsFromAppContainer()`](https://github.com/maidsafe/safe-getting-started-android/blob/master/app/src/main/java/net/maidsafe/sample/services/SafeApi.java#L58) function in the `SafeApi` class. This function is called after establishing a connection to the SAFE network.
+
+If you now run the application it will perform the mutable data operations that we have implemented via the APIs.
+
+## Migrating to the Alpha-2 network
```

Comment:
Could have an explanation under this - looks like its an empty heading. Alternatively should be a bigger font than the sections below it to emphasise it's a parent